api/python/quilt3/backends/local.py

```diff
@@ -20,18 +22,52 @@ def list_packages(self):
             for pkg in safe_listdir(os.path.join(pointers_dir_path, usr)):
                 yield f'{usr}/{pkg}'
 
-    def list_package_versions(self, pkg_name: str):
+    def list_package_pointers(self, pkg_name: str):
         pointers_dir_path = self.pointers_dir(pkg_name).path
         for path in safe_listdir(pointers_dir_path):
             with open(os.path.join(pointers_dir_path, path)) as f:
                 yield path, f.read()
 
-    def delete_package(self, pkg_name: str, top_hash: str = None):
-        package_path = self.pointers_dir(pkg_name)
-        if top_hash is None:
-            shutil.rmtree(package_path.path)
-        else:
-            super().delete_package(pkg_name, top_hash)
-            delete_url(package_path)
-        usr = pkg_name.partition('/')[0]
-        delete_url(self.pointers_global_dir.join(usr))
+    def delete_package(self, pkg_name: str):
+        shutil.rmtree(self._pointers_usr_dir(pkg_name).path)
+
+    def delete_package_version(self, pkg_name: str, top_hash: str):
+        super().delete_package_version(pkg_name, top_hash)
+        delete_url(self.pointers_dir(pkg_name))
+        delete_url(self._pointers_usr_dir(pkg_name))
+
+
+class LocalPackageRegistryV2(PackageRegistryV2):
+    _from_path_to_package_name = operator.methodcaller('replace', '@', '/')
```

Comment:
This could be a good opportunity to drop the <user>/<name> requirement for package names.